horovod/common/gloo_context.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Uber Technologies, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ============================================================================
+
+#include "gloo_context.h"
+
+#include "gloo/mpi/context.h"
+#include "gloo/transport/tcp/device.h"
+
+namespace horovod {
+namespace common {
+
+void GlooContext::InitializeFromMPI(const MPI_Comm &mpi_comm, const char*
+gloo_iface) {
+  gloo::transport::tcp::attr attr;
+  attr.iface = gloo_iface;
```

Comment:
What happens if we have two TCP interfaces that we should load balance across?

Reply:
Currently, we are hard-coding the interface either specified by the user or `eth0` by default. Is load balancing a feature that we need to support?

Reply:
After looking into the gloo code, it doesn't support multiplexing. Maybe we can report an issue to gloo and wait for their updating.

Reply:
FYI: https://github.com/facebookincubator/gloo/issues/183

Reply:
Please add a comment linking to the issue above, e.g.:

`TODO(sihan): add interface load balancing after https://github.com/facebookincubator/gloo/issues/183 is resolved

Reply:
Added.